client/src/app/shared/components/sidebar/sidebar.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { AuthService } from '../../../services/auth.service';
+import { Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-sidebar',
+  templateUrl: './sidebar.component.html',
+  styleUrls: ['./sidebar.component.css']
+})
+export class SidebarComponent implements OnInit, OnDestroy {
+  isAuthenticated = false;
+  isCollapsed = false;
+  private destroy$ = new Subject<void>();
+
+  constructor(private authService: AuthService, private router: Router) {}
+
+  ngOnInit(): void {
+    this.authService.authStatus$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((status) => {
+        this.isAuthenticated = status;
+      });
```

Comment:
```javascript
 this.authService.authStatus$
      .pipe(takeUntil(this.destroy$))
      .subscribe((status) => {
        this.isAuthenticated = status;
 });
``` 

You can shorten this operation by using `| async` pipe. So, for that you will need to make in
 `constructor(public authService: AuthService, private router: Router) {}`
 
`authService` as `public` to be able to use it in HTML, than in HTML you can write like this: 
 
 ```HTML
 <nav class="sidebar bg-light" *ngIf="authService.authStatus$ | async" [ngClass]="{'sidebar-collapsed': isCollapsed}">
 ```
 
 The benefits of this are: 
  1. `| async` pipe automatically subscribes AND **unsubscribes**  from your Observable, so you don't need to handle it's unsubscr. in `ngOnDestroy` method
  2. Clear and more readable code in HTML
  3. Less code in `.ts` files
 
 And you have to just remove the whole `ngOnInit`, `ngOnDestoy` method + implementation from the `class`, `destroy$` variable, as you don't need to handle unsubscription any more.

Reply:
Fixed